### PR TITLE
Refactoring Gradle configuration

### PR DESCRIPTION
**Pull request by BorzdeG**
* Refactoring Gradle configuration
* Upgraded Gradle Wrapper and fixed permissions
* Added detekt
* Upgraded Bintray plugin

### CODE REVIEW

**Comment by BorzdeG** [comment]
Do not accept while PR - I want to improve the code

**Comment by BorzdeG** [comment]
Done.
It is possible to build in Bintray publish? I did not find it [here](https://bintray.com/microutils)...

File: src/main/kotlin/kmu/KMU.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package kmu
 
 inline fun <T> T.match(predicate: (T) -> Boolean) = predicate(this)
 
-inline fun <T> XXX(): T = throw NotImplementedError()
+@Suppress("FunctionName")
+fun <T> XXX(): T = throw NotImplementedError()
```
Comment (oshai) [inline src/main/kotlin/kmu/KMU.kt]: why did you remove the inline?

**Review by oshai — CHANGES_REQUESTED** [review]

File: src/main/kotlin/kmu/KMU.kt (hunk 1 above)
Comment (BorzdeG) [reply src/main/kotlin/kmu/KMU.kt]: I was guided by this:
* compiler warning: `Expected performance impact of inlining 'public inline fun <T> XXX(): T defined in kmu in file KMU.kt' can be insignificant. Inlining works best for functions with lambda parameters`
* [When we don’t have function type parameter, reified type parameter, and we don’t need non-local return, then we most likely shouldn’t use inline modifier.](https://blog.kotlin-academy.com/effective-kotlin-consider-inline-modifier-for-higher-order-functions-758afcaffc11) 
* [KT-21020](https://youtrack.jetbrains.com/issue/KT-21020)

But, having looked at how the [TODO()](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/util/Standard.kt#L22) method was declared in Kotlin, I think that, most likely, I was wrong with XXX :(
return inline?

**Comment by oshai** [comment]
yes, I think if it is one statement it is better to have it inlined anyway.
specially here that exception is thrown.

**Review by oshai — APPROVED** [review]

**Comment by oshai** [comment]
Are you planning more changes before creating the package?
Have you tested the gradle configuration?

**Comment by BorzdeG** [comment]
In the near future, I do not plan changes.
Yes, I checked on the tasks "check" and "publishToMavenLocal"

**Comment by oshai** [comment]
I am sorry, didn't get to it yet, hope to do that soon.

**Comment by oshai** [comment]
@BorzdeG as time passes by - I think I will not publish it to maven as it is not on the top of my list right now. feel free to fork the lib and do it by yourself in your own bintray account.
